src/orion/core/io/cmdline_parser.py

```diff
@@ -0,0 +1,229 @@
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.core.io.cmdline_parser` -- Parse command line arguments for Orion
+=============================================================================
+
+.. module:: cmdline_parser
+   :platform: Unix
+   :synopsis: Parsing and building of the command line input for using script
+
+Simplify the parsing of a command line by storing every values inside an `OrderedDict`
+mapping the name of the argument to its value as a key-pair relation. Positional arguments
+are stored in the format `_X` where `X` represent the index of that argument inside
+the command line.
+
+CmdlineParser provides an interface to parse command line arguments from input but also
+facilities to build it again as a list or an already formatted string.
+"""
+from collections import OrderedDict
+import os
+
+
+class CmdlineParser(object):
+    """Simple class for commandline arguments parsing.
+
+    CmdlineParser aims at providing a simple class to interpret commandline arguments
+    for the purposes of Orion. It can transform a list of string representing arguments to their
+    corresponding values. It can also recreate that string from the values by maintaing a template
+    of the way the arguments were passed.
+
+    Parameters
+    ----------
+    parsing_tokens : dict, optional
+        Dictionary containing the tokens to use when parsing `-` and `_`.
+
+    Attributes
+    ----------
+    arguments : OrderedDict
+        Commandline arguments' name and value(s).
+    template : list
+        List of template-ready strings (in the form 'something_{0}') for formatting.
+
+    """
+
+    def __init__(self, parsing_tokens=None):
+        """See `CmdlineParser` description"""
+        self.arguments = OrderedDict()
+        self._already_parsed = False
+        self.template = []
+
+        parsing_tokens = {'-': '??', '_': '!!'}
+
+        if parsing_tokens is not None:
+            parsing_tokens.update(parsing_tokens)
+
+        self._dash_token = parsing_tokens['-']
+        self._underscore_token = parsing_tokens['_']
+
+    def format(self, configuration):
+        """Format the current template.
+
+        Recreate the string of argument using the template made at parse-time and
+        the values inside the `configuration` argument.
+
+        Parameters
+        ----------
+        configuration : dict
+            Dictionary storing the keys and values to be passed to the `format` function.
+            This would typically be `parser.arguments` where `parser` is a `CmdlineParser`
+            instance.
+
+        Returns
+        -------
+        str
+            A recreated string of the commandline passed to Orion.
```

Comment:
Of a commandline in general, not specific to Oríon.

Reply:
What about
```
The string of the original commandline parsed by the CmdlineParser, but formatted with given `configuration`.
```

Also, if you could add an example taking inspiration from the unit-tests that would be nice.

Reply:
Will do.

Reply:
Bump

Reply:
This change is inside unpushed commits.